src/Repository/CreditRepository.php

```diff
@@ -0,0 +1,73 @@
+<?php
+namespace App\Repository;
+
+use Wikimedia\ToolforgeBundle\Service\ReplicasClient;
+
+class CreditRepository {
+
+	public function getDatabaseName( $domain, $family ) {
+		if ( $family === 'wikisource' ) {
+			if ( in_array( $domain, [ 'old', 'www', 'mul' ] ) ) {
+				$dbname = 'sourceswiki_p';
+			} elseif ( $domain === 'zh-min-nan' ) {
+				$dbname = 'zh_min_nanwikisource_p';
+			} else {
+				$dbname = $domain . $family . '_p';
+			}
+		} else {
+			$dbname = $domain . $family . '_p';
+		}
+		return $dbname;
+	}
+	
+	public function getPageCredits( ReplicasClient $replicasClient, $lang, $namespaces, $pages ) : array {
+		$dbname = $this->getDatabaseName( $lang, 'wikisource' );
+
+		$conn = $replicasClient->getConnection( $dbname );
+		$pageNames = [];
+		$namespaceName = "";
+		$splittedPages = [];
+	
+		foreach ( $pages as $page ) {
+			$splitPage = explode( ":", $page, 2 );
+			if ( isset( $splitPage[1] ) ) {
+				$namespaceName = $splitPage[0];
+				$pageNames[] = $splitPage[1];
+				$namespaceId = $namespaces[$namespaceName];
+				$splittedPages[$namespaceId][] = "'" . $splitPage[1] . "'";
+			} else {
+				$pageNames[] = $splitPage[0];
+				$splittedPages[0][] = "'" . $splitPage[0] . "'";
+			}
+		}
+
+		$whereSql = [];
+
+		foreach ( $splittedPages as $namespaceId => $pages ) {
+			$whereSql[] = "( page_namespace = $namespaceId AND page_title IN (" . implode( ",", $pages ) . ")" . ")";
+		}
+		$whereSqlString = "(" . implode( ' OR ', $whereSql ) . ")";
+
+		$stmt = $conn->prepare( "SELECT
```

Comment:
At this point we're not actually preparing anything, so you could use `$conn->executeQuery( ...sql... )->fetchAll()`. Typically you do want to use prepared statements to guard against SQL injection, but I think Doctrine's `Connection::quote()` method will do that work, as suggested above on line 37

Reply:
Done!